### DISCUSSION TITLE
Any opinions on next-plugin-preval?

### DISCUSSION THREAD

**Discussion by zanami** [blitz-js/blitz#2720] [General]
Hey

Anyone using https://github.com/ricokahler/next-plugin-preval with Blitz??

I'm building an app with some mostly static data which comes from the DB (3-5 lists of 5 to 50 simple records) and it's used everywhere.
I'm evaluating Preval now, it can fetch and store the data at build time and then I can either:

1. Import as objects (probably fastest but requires code modifications)
2. Make Blitz queries to respond with JSON.parse("my-static-data-object") using Preval

I've tried the latter and it works. It still requires API call but eliminates DB queries at runtime.

Using Preval  doesn't feel very convenient  and I still can't decide where to put all the .prevaj.js stuff. But it works

**Comment by MrLeebo** [comment]
Could also be a use case for IndexedDB, so you can keep the data mostly static, but leave the door open so you can update it without necessarily triggering a new deployment of the app

**Reply by zanami** [reply]
I'm not familiar with IndexedDB, it looks like client-side storage, so it's basically a local cache which needs to be invalidated somehow. I'd prefer [persistQueryClient](https://react-query.tanstack.com/plugins/persistQueryClient) for that matter. Not sure how to plug it in a Blitzjs app.
